checker/tests/nullness/RepeatKeyFor.java

```diff
@@ -0,0 +1,77 @@
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.EnsuresKeyFor;
+import org.checkerframework.checker.nullness.qual.EnsuresKeyForIf;
+
+public class RepeatKeyFor {
+
+    Map<String, Integer> map = new HashMap<>();
+    Map<String, Integer> map2 = new HashMap<>();
+
+    public void func1(String a, String b, String c) {
+        map.put(a, 1);
+        map.put(c, 2);
+        map2.put(a, 3);
+    }
+
+    public boolean func2(String a, String b, String c) {
+        map.put(a, 1);
+        map.put(c, 2);
+        map2.put(b, 3);
+        return true;
+    }
+
+    @EnsuresKeyFor(
+            value = {"#1", "#2"},
+            map = "map")
+    @EnsuresKeyFor(value = "#2", map = "map2")
+    // :: error:  (contracts.postcondition.not.satisfied)
+    public void samefunc1(String a, String b, String c) {
```

Comment:
"same" is not very illuminating as a name.  Even "with postconditions" would be clearer.

Reply:
Fixed!